cmd/root.go

```diff
@@ -96,11 +96,14 @@ func init() {
 	})
 
 	audit.Logger = audit.New(pkg.Options.AuditFile)
+
 }
 
 func Execute() {
+	defer audit.Delete(pkg.Options.AuditFile)
 	if err := rootCmd.Execute(); err != nil {
 		klog.Errorln(err)
+		audit.Delete(pkg.Options.AuditFile)
```

Comment:
why are we calling this again if its already mentioned above as defer?

Reply:
As Defer will not be executed after os.Exit. we are calling the Delete explicitly before that.

Reply:
hmm..., lets handle the exit in the main() by returning err from here

Reply:
Done.